### ISSUE TITLE
misnamed-import warning for import A from '.'

### ISSUE THREAD

**Issue by devdoomari** [Microsoft/tslint-microsoft-contrib#378]
Here's a somewhat edge-case example of what I've stumbled on:

```typescript
// file path: <proj-path>/src/Logger/utils.ts
import Logger, {
  LogItem,
} from '.'; //this refers to <proj-path>/src/Logger/index.ts
```

Here's what I get:

```
error: Logger/logger.utils.ts
severity: 'Error'
message: 'Misnamed import. Import should be named '.' but found 'Logger' (import-name)'
at: '2,1'
source: 'tslint'
```

I wrote a tslint issue on this(https://github.com/palantir/tslint/issues/2878) but was told to write the issue here.

ps: I can do a walk-around using:

```typescript
import {
  default as Logger,
  ...
} from '.';
```

but I don't like doing tricks to please Mr Linter :(

**Comment by osdiab** [comment]
I also experienced this: via

```
import Home from "./"; // lint error: Misnamed import. Import should be named '' but found 'Home' (import-name)
```

**Comment by osdiab** [comment]
I tried a quick stab at this but then I realized that I'm not sure what a good way to reliably get a module's name is. [The code](https://github.com/Microsoft/tslint-microsoft-contrib/blob/master/src/importNameRule.ts#L86) just takes anything after the final `/` character as the module name, which explains our errors, but it doesn't actually resolve relative paths or anything so a path like `./` wouldn't have any way of figuring out what directory that import was in.

I don't know enough about how typescript works to know how to figure out the module name despite this, so if someone knows more that would be great!

**Comment by sindresorhus** [comment]
The rule should just ignore imports of the format `.`, `./`, `./index`, for now, until its logic is improved to resolve the actual names. Right now, the rule is completely unusable.

**Comment by Dgaduin** [comment]
If this is still up, I can have a go at it.

**Comment by Igorbek** [comment]
@Dgaduin oops, sorry I didn't notice you grabbed it. I'm not sure my fix is correct though.

**Comment by dougmolineux** [comment]
This seems to be still broken, using "tslint-microsoft-contrib": "^6.1.1",

> Misnamed import. Import should be named '' but found 'serviceBusTopicTrigger' tslint(import-name)

`import serviceBusTopicTrigger from './';`

It's exported like this:

`export default serviceBusTopicTrigger;`

What am I doing wrong?

**Comment by IllusionMH** [comment]
@dougmolineux could you please create separate issue using full template (with all versions specified)? It was really easy to miss comment on closed issue and I'm unable to reproduce problem on version `6.1.1` and `6.2.0`. Fix is in place https://github.com/microsoft/tslint-microsoft-contrib/blob/bfc28e95dde1fab92aa457e79d338fb03b81d7cc/src/importNameRule.ts#L249-L252

**Comment by dougmolineux** [comment]
@IllusionMH sure thing, will do.
